codex/conf.nim

```diff
@@ -293,6 +293,22 @@ type
         name: "validator-max-slots"
       .}: int
 
+      validatorPartitionSize* {.
+        desc: "Slot validation partition size"
+        longDesc: "A number indicating total number of partitions into which the whole slot id space will be divided. " &
+          "The value must be greater than 0"
+        defaultValue: 1
+        name: "validator-partition-size"
+      .}: int
+      
+      validatorPartitionIndex* {.
+        desc: "Slot validation partition index"
+        longDesc: "The value provided must be in the range [0, partitionSize). " &
+          "Only slot ids satisfying condition [(slotId mod partitionSize) == partitionIndex] will be observed by the validator"
+        defaultValue: 0
```

Comment:
What happens if I want to validate all the slots in the slot Id space, and let the `--validator-max-slots` limit the number (ie the current behaviour of the node prior to this PR)? In my opinion, we should use an `Option[int]` (or `Option[uint16]`, see below) for this type with `int.none` being default which indicates the node will validate all slots (and be limited by `--validator-max-slots`).

Reply:
I see, yes indeed using `1` for this parameter, which is default would satisfy this need. However, again my personal opinion (which absolutely does not mean it's right) is that it's more explicit in the code and easier to understand / follow if it's an `Option`, where `uint16.none` means the user supplied nothing, and we don't need to understand that `1` is "special". Personal preference again I guess.

Reply:
But the user still can provide `1`, so it will remain special. But let's talk it quickly through tomorrow.

Reply:
Sure, but if the user provides `1`, they know what they're doing :)

Reply:
And if they will not provide anything, it will default to `1` thus the "old" behaviour, so also good ;).

Reply:
Using optionals 👍